src/shared/bg_utilities.cpp

```diff
@@ -228,3 +228,127 @@ WARN_UNUSED_RESULT glm::mat3 RotationMatrix( const glm::vec3 &angles ) {
 	m[1] *= -1.0f;
 	return m;
 }
+
+glm::vec3 ProjectPointOntoVectorBounded(const glm::vec3 &point, const glm::vec3 &vStart, const glm::vec3 &vEnd)
+{
+	glm::vec3 pVec = point - vStart;
+	glm::vec3 vec = vEnd - vStart;
+	VectorNormalize( vec );
+	// project onto the directional vector for this segment
+	glm::vec3 vProj = vStart + glm::dot(pVec, vec) * vec;
+
```

Comment:
How about `return vStart + clamp(dot(pVec, vec), 0, 1) * vec`?

Reply:
Sounds good, but I'm not sure about corner cases, rounding errors, etc. I'd like to keep the original math for now.